app/services/proposal_service.rb

```diff
@@ -0,0 +1,55 @@
+class ProposalService
+  # include Rails.application.routes.url_helpers
```

Comment:
Does all the commented out code need to be here?

Reply:
Not sure why comments have been left in the code.